common/src/chain/transaction/output/timelock.rs

```diff
@@ -0,0 +1,15 @@
+use serialization::{Decode, Encode};
+
+use crate::{chain::block::timestamp::BlockTimestamp, primitives::BlockHeight};
+
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Encode, Decode)]
+pub enum OutputTimeLock {
```

Comment:
Why is this `OutputTimeLock`? Any other time lock kinds planned? It seems to me that it could be useful in script-enforced input time locks too.

Reply:
I'm personally not familiar with all timelock types, which is why I asked for your review. As far as I know, the problem with script timelocks is that they're bound to the integer time lock in the transaction... that's incompatible with this enum. Honestly I don't know how to reconcile all those. You're more familiar with this. If this is conclusively determined and doesn't conflict with anything, I'll be happy to change it to just Timelock. If my choice is not crazy (in the sense that we can update it in the near future), we can keep it. If there's an obvious mistake/conflict, let me know. I'm not sure after all.

Reply:
The issue with script-based time locks it that the absolute time lock is enforced via transaction-level time lock field and relative time lock is enforced via input-level time lock field (nSequence). If we follow that design, this `OutputTimeLock` type would not work so neatly, so let's leave it as it is at the moment and have a more in-depth discussion at some point.